### PR TITLE
Add abort functionality in fetch stream

### LINKED ISSUE #9698: PDF.js should abort fetch requests during teardown
Steps to reproduce the problem:
1. Fetch a document with streaming enabled.
2. Destroy the document loading task
3. Observe that the fetch request is not aborted.

What is the expected behavior?
Fetch request should be aborted when cancel is called on `PDFFetchStreamReader` and `PDFFetchStreamRangeReader`.

**Context**
It seems when fetch was implemented (https://github.com/mozilla/pdf.js/issues/6126#issuecomment-130572300) `abort` on fetch was still in flux. 

It seems now that has settled and chromium has also implemented aborting fetch requests in 66 (https://bugs.chromium.org/p/chromium/issues/detail?id=750599)

### PR DESCRIPTION
**Pull request by mukulmishra18**
Fixes #9698

### CODE REVIEW

**Comment by mukulmishra18** [comment]
I can see node `Aborted (core dump)` error with an error message:

`../src/node_contextify.cc:631:static void node::contextify::ContextifyScript::New(const v8::FunctionCallbackInfo<v8::Value>&): Assertion 'args[1]->IsString()' failed.` 

I think there is some issue with gulp and node v10.0.0. It is something related to internal contextify script of nodeJS. See https://github.com/nodejs/node/issues/20281 node issue for more details.

**Comment by mukulmishra18** [comment]
@Rob--W Can you please review this PR?

File: src/display/fetch_stream.js [hunk 1]
```diff
@@ -80,6 +81,10 @@ class PDFFetchStreamReader {
       this._disableRange = true;
     }
 
+    if (window.AbortController) {
```
Comment (Snuffleupagus) [inline src/display/fetch_stream.js]: Drive-by nit: It's probably best to *not* reference `window` in `/src` code unless actually necessary, so `typeof AbortController !== 'undefined'` or `typeof AbortController === 'function'` would seem like a much better way of feature-testing here.

File: src/display/fetch_stream.js [hunk 2]
```diff
@@ -184,6 +193,11 @@ class PDFFetchStreamRangeReader {
     this._readCapability = createPromiseCapability();
     this._isStreamingSupported = !source.disableStream;
 
+    if (window.AbortController) {
```
Comment (Snuffleupagus) [inline src/display/fetch_stream.js]: Drive-by nit: See https://github.com/mozilla/pdf.js/pull/9701/files#r189454633.

File: src/display/fetch_stream.js (hunk 1 above)
Comment (mukulmishra18) [reply src/display/fetch_stream.js]: Seems reasonable. Done in new commit. Thanks.

File: src/display/fetch_stream.js (hunk 2 above)
Comment (mukulmishra18) [reply src/display/fetch_stream.js]: Done.

File: src/display/fetch_stream.js [hunk 3]
```diff
@@ -80,6 +81,10 @@ class PDFFetchStreamReader {
       this._disableRange = true;
     }
 
+    if (window.AbortController) {
+      this._abortController = new AbortController();
+      this._abortSignal = this._abortController.signal;
```
[the remark this answers is not in the document]
Comment (mukulmishra18) [reply src/display/fetch_stream.js]: Passing `abortSignal` directly to `createFetchOptions` seems a good idea. Done in commit. Thank You.

File: src/display/fetch_stream.js [hunk 4]
```diff
@@ -171,6 +176,10 @@ class PDFFetchStreamReader {
     if (this._reader) {
       this._reader.cancel(reason);
     }
+    // Abort fetch if it is abortable.
```
[the remark this answers is not in the document]
Comment (mukulmishra18) [reply src/display/fetch_stream.js]: Removed in new commit. Thanks.

**Review by Rob--W — APPROVED** [review]
LGTM, thanks for the patch.
